ot2/octopus-full-ssh.py

```diff
@@ -0,0 +1,664 @@
+            self.p300.consolidate(5, [self.pre_pool_plate_2.rows_by_name()['B'][num_plate].bottom(z=0.5) for num_plate in range(0, 23, 2)], self.consolidation_plate.columns('7'))
+        if NUM_PLATES == 8:
+            self.p300.consolidate(5, [self.pre_pool_plate_2.rows_by_name()['B'][num_plate].bottom(z=0.5) for num_plate in range(1, 24, 2)], self.consolidation_plate.columns('8'))
+
+        ### further consolidate plate columns into single wells on the DNA pool deep well plate
+        self.set_flow_rate(48*2, 48, 48)
+        for i in range(NUM_PLATES):
+            self.p300.pick_up_tip(self.tips_200[0].rows()[7-i][NUM_PLATES])
+            # aspirate first four wells
+            for j in range(4):
+                self.p300.aspirate(50, self.consolidation_plate.rows()[j][i].bottom(z=0.5))
+            self.p300.dispense(200, self.pool_plate.rows()[i][0])
+            # aspirate last four wells
+            for j in range(4):
+                self.p300.aspirate(50, self.consolidation_plate.rows()[j+4][i].bottom(z=0.5))
+            self.p300.dispense(200, self.pool_plate.rows()[i][0])
+            self.p300.drop_tip()
+
+        ### add and incubate Purification Beads with pooled samples
+        self.p300.pick_up_tip()
+        self.set_flow_rate(48*4, 48*4, 48)
+        self.p300.mix(10, 200, Purification_Beads.bottom(z=10.0))
+            # adding beads to pool
+        self.set_flow_rate(48*2, 48*4, 48)
+        for i in range(3):
+            self.p300.aspirate(200, Purification_Beads)
+            self.p300.dispense(200, DNA_pool)
+        self.p300.aspirate(120, Purification_Beads)
+        self.p300.dispense(120, DNA_pool)
+            # mix beads well with pool and incubate
+        self.set_flow_rate(48*4, 48*6, 48)
+        self.p300.mix(10, 200, DNA_pool.bottom(z=10.0))
+        self.set_flow_rate(48*4, 48*4, 48)
+        for i in range(20):
+            self.p300.aspirate(200, DNA_pool.bottom(z=10.0))
+            self.p300.dispense(200, DNA_pool.center())
+        self.set_flow_rate(48*4, 48*6, 48/6)
+        self.p300.mix(20, 200, DNA_pool.bottom(z=10.0))
+        self.p300.move_to(DNA_pool.top())
+        self.protocol.delay(seconds=5.0)
+        self.p300.blow_out(DNA_pool.top())
+        self.p300.drop_tip()
+        self.protocol.delay(minutes=10)
+        self.mag_mod.engage()
+        self.protocol.delay(minutes=12)
+
+        ### discard supernatant in stages to minimize bead loss
+        self.set_flow_rate(48/2, 48, 48)
+        self.p300.pick_up_tip()
+        self.p300.aspirate(200, DNA_pool.bottom(z=0.5))
+        self.p300.dispense(200, Purification_Beads_discard)
+        self.p300.move_to(Purification_Beads_discard.top(z=10.0))
+        self.protocol.delay(minutes=4)
+        self.p300.aspirate(200, DNA_pool.bottom(z=0.5))
+        self.p300.dispense(200, Purification_Beads_discard)
+        self.p300.move_to(Purification_Beads_discard.top(z=10.0))
+        self.protocol.delay(minutes=2)
```

Comment:
Mmmm yes, it was uncomfortable leaving this section rolled out like this, and I would have it wrapped up in a loop-like structure if I had to otherwise. However, I thought having a method like 
`pool_stage_bead_supernant_discard(volume, source, destination, delay, has_delay=True, bottom=0.5, top=10.0)` 
in such a localized area seemed less readable that just having the area isolated by comment line breaks.

Reply:
```python
delays = [4, 2, 2, 2, 2, None]
for delay in delays:
 self.p300.aspirate(200, DNA_pool.bottom(z=0.5))
  self.p300.dispense(200, Purification_Beads_discard)
  self.p300.move_to(Purification_Beads_discard.top(z=10.0))
  if delay:
      self.protocol.delay(mintues=delay)
```